### PR TITLE
Use lists to handle tax function parameters throughout

### PR DESCRIPTION
**Pull request by jdebacker**
This PR fixes OG-Core to pass tax function parameters as lists throughout the model.  This will allow one to use the spline functions invoked with the `tax_func_type=mono` option (introduced in PR #819) and the `mono2D` tax functions (to be introduced in PR #839).

### CODE REVIEW

**Comment by jdebacker** [comment]
I believe this PR is ready to go. Still running the full suite of unit tests locally, but they are looking good.

Note, I tried to use list comprehensions in all places to avoid loops, but the code still seems to have slowed down a little bit since moving away from bumpy arrays.

File: ogcore/parameters.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import scipy.interpolate as si
 import pkg_resources
 import paramtools
+from itertools import chain
```
Comment (rickecon) [inline ogcore/parameters.py:6]: @jdebacker. I don't see this `chain` module used anywhere in `parameters.py`. If that is the case, we should remove this.

**Comment by rickecon** [comment]
@jdebacker. I have reviewed this code and it looks good to me. The stuff in `TPI.py` and `parameters.py` was particularly difficult. Thank you for doing that. I am just running all the local tests on my machine. It should be done tomorrow afternoon. Once those all pass (minus the tests you noted above), I'll merge this.

**Comment by rickecon** [comment]
@jdebacker. All the local tests ran on my machine last night in 8 hrs, 27 min, 13 sec (30,433.19s). I got the same test failures as you, with one additional test failure in `test_SS.py::test_run_SS[Reform, use zeta]'. The error is Steady-state equilibrium not found.
```
============================= test session starts ==============================
platform darwin -- Python 3.10.9, pytest-7.2.1, pluggy-1.0.0
rootdir: /Users/richardevans/Docs/Economics/OSE/OG-Core, configfile: pytest.ini
plugins: xdist-3.2.0
collected 492 items                                                            

tests/test_SS.py .......................F.....                           [  5%]
tests/test_TPI.py ............................                           [ 11%]
tests/test_aggregates.py .....................................           [ 19%]
tests/test_basic.py ....                                                 [ 19%]
tests/test_elliptical_u_est.py .......                                   [ 21%]
tests/test_execute.py .                                                  [ 21%]
tests/test_firm.py ..................................................... [ 32%]
.............                                                            [ 34%]
tests/test_fiscal.py ...................                                 [ 38%]
tests/test_household.py ..............................................   [ 48%]
tests/test_output_plots.py ............................................. [ 57%]
                                                                         [ 57%]
tests/test_output_tables.py .............                                [ 59%]
tests/test_parameter_plots.py ....................................       [ 67%]
tests/test_parameter_tables.py .......                                   [ 68%]
tests/test_parameters.py .............                                   [ 71%]
tests/test_run_example.py ..                                             [ 71%]
tests/test_run_ogcore.py .                                               [ 71%]
tests/test_tax.py ......................................                 [ 79%]
tests/test_txfunc.py .....F.......F..........F.                          [ 84%]
tests/test_user_inputs.py .........                                      [ 86%]
tests/test_utils.py .................................................... [ 97%]
.............                                                            [100%]
===== short test summary info ============
FAILED tests/test_SS.py::test_run_SS[Reform, use zeta] - RuntimeError: Steady state equilibrium not found
FAILED tests/test_txfunc.py::test_txfunc_est[DEP] - assert False
FAILED tests/test_txfunc.py::test_tax_func_loop - assert False
FAILED tests/test_txfunc.py::test_tax_func_estimate - assert False
====== 4 failed, 488 passed, 65 warnings in 30433.19s
```
The `test_SS.py` failure looks like it went through all the possible starting values and still didn't solve. If this test did not fail on your local tests, then I am OK merging this.

**Comment by jdebacker** [comment]
`tests/test_SS.py::test_run_SS[Reform, use zeta]` did not fail on my local testing.

**Review by rickecon — APPROVED** [review]

**Comment by rickecon** [comment]
@jdebacker. OK then. Looks good to me. Merging.

**Pull request merged**